lib/vdc/CurvilinearGrid.cpp

```diff
@@ -237,14 +237,14 @@ CurvilinearGrid::ConstCoordItrCG::ConstCoordItrCG(const CurvilinearGrid *cg, boo
         _index[ndims - 1] = dims[ndims - 1];
         return;
     }
-    _coords.push_back(*_xCoordItr);
-    _coords.push_back(*_yCoordItr);
+    _coords[0] = *_xCoordItr;
+    _coords[1] = *_yCoordItr;
 
     if (ndims == 3) {
         if (_terrainFollowing) {
-            _coords.push_back(*_zCoordItr);
+            _coords[2] = *_zCoordItr;
         } else {
-            _coords.push_back(_cg->_zcoords[0]);
+            _coords[2] = _cg->_zcoords[0];
         }
     }
```

Comment:
Should give _coords[2] a default value in condition of ndims != 3, right?

Reply:
Fixed